importApp/src/test/java/com/commercetools/dataimport/joyrideavailability/AvailabilityPricesImportJobIntegrationTest.java

```diff
@@ -0,0 +1,141 @@
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.commercetools.dataimport.joyrideavailability.AvailabilityPricesImportJobConfiguration.*;
+import static com.commercetools.dataimport.joyrideavailability.JoyrideAvailabilityUtils.*;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@IntegrationTest
+@ContextConfiguration(classes = {TestConfiguration.class, AvailabilityPricesImportJobConfiguration.class, CommercetoolsTestConfiguration.class})
+@TestPropertySource("/test.properties")
+@EnableAutoConfiguration
+@Configuration
+public class AvailabilityPricesImportJobIntegrationTest extends JoyrideAvailabilityIntegrationTest {
+
+    @Autowired
+    private JobLauncherTestUtils jobLauncherTestUtils;
+
+    @Autowired
+    private Environment env;
+
+    @Test
+    public void findLastProductWithJoyridePrice() throws Exception {
+        final Product product = createProduct(sphereClient);
+        final Long productsCount = 1L;
+        assertThat(findLastProductWithJoyrideChannel(sphereClient, 0L, productsCount - 1)).isEmpty();
+        withPriceWithJoyrideChannel(sphereClient, priceDraft -> {
+            final AddPrice addPrice = AddPrice.of(product.getMasterData().getCurrent().getMasterVariant().getId(), priceDraft);
+            Product updatedProduct = sphereClient.executeBlocking(ProductUpdateCommand.of(product, asList(addPrice, Publish.of())));
+            final Optional<ProductProjection> lastProductWithJoyrideChannel = findLastProductWithJoyrideChannel(sphereClient, 0L, productsCount - 1);
+            assertThat(lastProductWithJoyrideChannel).isPresent();
+            assertThat(lastProductWithJoyrideChannel.get().getMasterVariant().getSku()).isEqualTo(product.getMasterData().getCurrent().getMasterVariant().getSku());
+            final Product productToDelete = sphereClient.executeBlocking(ProductUpdateCommand.of(updatedProduct, Unpublish.of()));
+            sphereClient.executeBlocking(ProductDeleteCommand.of(productToDelete));
+        });
+    }
+
+    @Test
+    public void jobAvailabilityPricesImport() throws Exception {
+        final int amountOfProducts = 10;
+        withJoyrideChannels(sphereClient, joyrideChannels -> {
+            withListOfProductProjections(sphereClient, amountOfProducts, productsWithoutJoyride -> {
+                assertThat(productsWithoutJoyride.size()).isEqualTo(amountOfProducts);
+                executeAvailabilityPricesImportJob();
+                final List<ProductProjection> updatedProducts = sphereClient.executeBlocking(ProductProjectionQuery.ofCurrent().withExpansionPaths(m -> m.allVariants().prices().channel())).getResults();
+                updatedProducts.forEach(m -> {
```

Comment:
I would not use `m` as variable name if its scope of existence so long.

Maybe you could rename it to sth. like `updatedProductProjection`